# Hugen2072_coding_assignment
Review coding assignment utilizing Unix, PLINK, VCF/BCF, and R/RMarkdown

## Files required for submission:

All are in the root of this repository:
- Your slurm script file for steps 1–10 -> ca.sh
- The slurm log file for the script when you ran the script -> Log_ca.sh.out
- The RMarkdown file for steps 11–20 -> ca.Rmd
- The knitted HTML file produced by the RMarkdown file -> ca.html (also included ca.md to allow viewing on GitHub)

## CRC/bash/Slurm component: 

1. Create and run a script on the CRC cluster that 

2. Copies data.vcf to your working directory, 

3. Then uses bcftools to create a data4.bcf.gz file that is sorted and indexed, and 

4. Is filtered to include only positions on chromosome 4; 

5. Then uses PLINK to create a PLINK binary file set version of data4.bcf.gz called data4.{fam,bim,bed}; 

6. Then uses PLINK to update sex variable in the data4.fam file using the sex variable in sex.txt (without copying sex.txt to your own directory), 

7. Then uses PLINK, coupled with the phenotypes in phenotype.txt (without copying phenotype.txt to your own directory—and you should use phenotype.txt as an auxiliary file for the next few tasks without altering data4.fam to include phenotype data), to 

8. Calculate the allele frequencies of the markers in data4.{fam,bim,bed} in the cases only and 

9.  Calculate the allele frequencies of the markers in data4.{fam,bim,bed} in the controls only; then 

10. Performs a GWAS of the phenotype using logistic regression with no covariates. 

## RMarkdown portion of the assignment: 

11. Then, create an RMarkdown file that knits to an HTML file which 

12. Reads the two allele frequency files and the logistic regression output into R, and 

13. Plots a histogram of the case allele frequencies, 

14. Plots a histogram of the control allele frequencies, 

15. Plots a scatterplot of the case individuals’ allele frequencies (y axis) 

16. Versus the control individuals’ allele frequencies (x axis), 

17. Plots a scatterplot of −log₁₀(p values) (y axis) versus chr4 basepair position (x axis), 

18. Reports the marker and p value of the marker with the lowest p value; then 

19. Loads the snp_annot.RData workspace and merges the p values into the SNP annotation data frame object snp_annot_df in such a way that 

20. The order of the SNPs in snp_annot_df is unchanged. 

 
